src/test/java/com/microsoft/sqlserver/jdbc/tvp/TVPAllTypesTest.java

```diff
@@ -169,13 +167,12 @@ public void testTVPDataTable() throws SQLException {
                     .prepareStatement("INSERT INTO " + tableDest.getEscapedTableName() + " select * from ? ;")) {
                 pstmt.setStructured(1, tvpName, dt);
                 pstmt.execute();
-            } finally {
-                terminateVariation(stmt);
             }
+            terminateVariation(stmt);

```

Comment:
move to finally

Reply:
I get your point to ensure tables are cleared. The current case threw random errors of Connection is closed. I pushed another update to use dedicated connection for terminateVariation to avoid such errors.